src/api/client.ts

```diff
@@ -0,0 +1,21 @@
+import { client } from '../utils/fetchClient';
+
+export const getUsers = () => {
+  return client.get('/users');
+};
+
+export const getUserPosts = (userId: number) => {
+  return client.get(`/posts?userId=${userId}`);
+};
+
+export const getPostComments = (postId: number) => {
+  return client.get(`/comments?postId=${postId}`);
+};
+
+export const postComment = (data: {}) => {
```

Comment:
Consider using a specific type for the data parameter instead of the generic object type. This will help catch any potential issues with the data structure at compile time.